cbr-service-profile/main.tf

```diff
@@ -0,0 +1,108 @@
+data "ibm_iam_account_settings" "iam_account_settings" {
+}
+
+##############################################################################
+#
+# CBR Rule for a list of target services
+##############################################################################
+locals {
+  # tflint-ignore: terraform_unused_declarations
+  validate_zone_inputs = ((length(var.zone_vpc_crn_list) == 0) && (length(var.zone_service_ref_list) == 0)) ? tobool("Error: Provide a valid zone vpc and/or service references") : true
+
+  # Restrict and allow the api types as per the target service
+  icd_api_types = ["crn:v1:bluemix:public:context-based-restrictions::::api-type:data-plane"]
+  operations_apitype_val = {
+    databases-for-enterprisedb  = local.icd_api_types,
+    containers-kubernetes       = ["crn:v1:bluemix:public:containers-kubernetes::::api-type:cluster", "crn:v1:bluemix:public:containers-kubernetes::::api-type:management"],
+    databases-for-cassandra     = local.icd_api_types,
+    databases-for-elasticsearch = local.icd_api_types,
+    databases-for-etcd          = local.icd_api_types,
+    databases-for-mongodb       = local.icd_api_types,
+    databases-for-postgresql    = local.icd_api_types,
+    databases-for-redis         = local.icd_api_types,
+    messages-for-rabbitmq       = local.icd_api_types
+  }
+
+  vpc_zone_list = (length(var.zone_vpc_crn_list) > 0) ? [{
+    name             = "${var.prefix}-cbr-vpc-zone"
+    account_id       = data.ibm_iam_account_settings.iam_account_settings.account_id
+    zone_description = "cbr-vpc-zone-terraform"
+    addresses = [
+      for zone_vpc_crn in var.zone_vpc_crn_list :
+      { "type" = "vpc", value = zone_vpc_crn }
+    ]
+  }] : []
+
+  service_ref_zone_list = (length(var.zone_service_ref_list) > 0) ? [{
+    name             = "${var.prefix}-cbr-serviceref-zone"
+    account_id       = data.ibm_iam_account_settings.iam_account_settings.account_id
+    zone_description = "cbr-serviceref-zone-terraform"
+    # when the target service is containers-kubernetes or any icd services, context cannot have a serviceref
+    addresses = [
+      for serviceref in var.zone_service_ref_list : {
+        type = "serviceRef"
+        ref = {
+          account_id   = data.ibm_iam_account_settings.iam_account_settings.account_id
+          service_name = serviceref
+        }
+      }
+    ]
+  }] : []
+  zone_list = concat(tolist(local.vpc_zone_list), tolist(local.service_ref_zone_list))
+}
+
+module "cbr_zone" {
+  count            = length(local.zone_list)
+  source           = "../cbr-zone-module"
+  name             = local.zone_list[count.index].name
+  zone_description = local.zone_list[count.index].zone_description
+  account_id       = data.ibm_iam_account_settings.iam_account_settings.account_id
+  addresses        = local.zone_list[count.index].addresses
+}
+
+locals {
+  rule_contexts = [{
+    attributes = [
+      {
+        "name" : "endpointType",
+        "value" : "private"
```

Comment:
Just checking do we always want this to be private? probably...

Reply:
This will create one context with all the zones in it. The other approach is to create a context per zone. Have you thought about the pros and cons of each approach (If any...) I think what you have is fine but I haven't looked into it much other than you can do it.

Reply:
Looking at the number of context a service team might have, having one context per zone may be cluttered.  On the other side since having the one context or multiple context has the same impact, I feel this is good enough right now.  If we get any specific feedback then we can discuss further to update.